rasa/graph_components/providers/nlu_training_data_provider.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, Optional
+import os
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.importers.importer import TrainingDataImporter
+from rasa.shared.nlu.training_data.training_data import (
+    TrainingData,
+    DEFAULT_TRAINING_DATA_OUTPUT_PATH,
+)
+from rasa.shared.nlu.training_data.loading import load_data
+
+
+class NLUTrainingDataProvider(GraphComponent):
+    """Provides NLU training data during training and inference time."""
+
+    def __init__(
+        self,
+        model_storage: ModelStorage,
+        resource: Resource,
+        training_data: Optional[TrainingData] = None,
+    ) -> None:
+        """Creates NLU training data provider."""
+        self._model_storage = model_storage
+        self._resource = resource
+        self._training_data = training_data
+
+    @classmethod
+    def get_default_config(cls) -> Dict[Text, Any]:
+        return {"language": "en", "persist": False}
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> NLUTrainingDataProvider:
+        """Creates component (see parent class for full docstring)."""
+        return cls(model_storage, resource)
+
+    @classmethod
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+        **kwargs: Any,
+    ) -> NLUTrainingDataProvider:
+        """Creates provider using a persisted data."""
+        with model_storage.read_from(resource) as resource_directory:
+            training_data = load_data(
+                resource_name=os.path.join(
+                    str(resource_directory), DEFAULT_TRAINING_DATA_OUTPUT_PATH
+                ),
+                language=config["language"],
+            )
```

Comment:
I modelled my implementation on `DomainProvider`. However, unlike the `Domain` class `TrainingData` did not have a `from_file` method. Did I use the correct nlu training data loader?

Reply:
So this is a weird one... but we don't actually need a `load` for this component!
We only persist to keep the legacy functionality of `--persist-nlu-data` which saves the nlu data with the model. We would never actually load this data.